examples/puppet/modules/install_otel_collector/manifests/init.pp

```diff
@@ -18,58 +18,119 @@
 #
+
 class install_otel_collector (
   String $installation_token,
   Hash[String, String] $collector_tags = {},
   Optional[String] $api_url = undef,
   Boolean $systemd_service = true,
   Optional[String] $version = undef,
   String $src_config_path = 'puppet:///modules/install_otel_collector/conf.d',
+  Optional[String] $opamp_api_url = undef,
 ) {
-  $install_script_url = 'https://download-otel.sumologic.com/latest/download/install.sh'
-  $install_script_path = '/tmp/install.sh'
-  $download_timeout = 300
+  if $facts['os']['family'] == 'windows' {
 
-  # construct the install command arguments from class parameters
-  $tags_command_args = $collector_tags.map |$key, $value| { "--tag ${key}=${value}" }
-  if $version == undef {
-    $version_command_args = []
-  } else {
-    $version_command_args = ["--version ${version}"]
-  }
-  if $api_url == undef {
-    $api_command_args = []
-  } else {
-    $api_command_args = ["--api ${api_url}"]
-  }
-  if $systemd_service {
-    $systemd_command_args = []
+    $install_script_url  = 'https://download-otel.sumologic.com/latest/download/install.ps1'
+    $install_script_path = 'C:/Windows/Temp/install.ps1'
+    $exe_path            = 'C:/Program Files/Sumo Logic/OpenTelemetry Collector/bin/otelcol-sumo.exe'
+
+    # Construct PowerShell hashtable literal string for tags
+    $tags_ps_lines = $collector_tags.map |$k, $v| {
+      "    '${k}' = '${v}';"
+    }
+    $tags_ps_block = "@{\n${tags_ps_lines.join("\n")}\n}"
+
+    # Optional arguments as strings (empty if undef)
+    $api_arg    = $api_url ? { undef => '', default => "-Api '${api_url}'" }
+    $opamp_arg  = $opamp_api_url ? { undef => '', default => "-OpAmpApi '${opamp_api_url}'" }
+    # Download install.ps1 script
+    exec { 'Download Otel Collector Script':
+      command   => "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -ExecutionPolicy Bypass -Command \"Invoke-WebRequest -Uri '${install_script_url}' -OutFile '${install_script_path}'\"",
+      creates   => $install_script_path,
+      logoutput => true,
+    }
+
+    # Run install script with tags defined inside PowerShell command
+    exec { 'Install Otel Collector':
+      command => "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -ExecutionPolicy Bypass -Command \"\
+        Set-ExecutionPolicy RemoteSigned -Scope Process -Force; \
+        [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; \
+        \$tags = ${tags_ps_block}; \
+        & '${install_script_path}' -InstallationToken '${installation_token}' -Tags \$tags ${api_arg} ${opamp_arg}\"",
+      unless  => "C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe -Command \"if (Test-Path '${exe_path}') { exit 0 } else { exit 1 }\"",
+      require => Exec['Download Otel Collector Script'],
+      logoutput => true,
+    }
+
+    # Ensure directories exist
+    file { 'C:/Program Files/Sumo Logic':
```

Comment:
Do we really need these  ensure  => directory , 4 times ?
maybe just check once like

    file { 'C:/Program Files/Sumo Logic/OpenTelemetry Collector/conf.d':
      ensure  => directory,
      require => File['C:/Program Files/Sumo Logic/OpenTelemetry Collector'],
    }